extensions/console/02_console.js

```diff
@@ -1958,6 +1962,60 @@
     });
   }
 
+  /** Creates a proxy that represents a subset of the properties
+   * of the original object optionally without evaluating the properties
+   * in order to get the values. */
+  function createFilteredInspectProxy({ object, keys, evaluate }) {
```

Comment:
This was taken and improved from my PR in #10979. I'm not sure about it existing in this file though and could use some guidance on what would be best for us to do here. On a related note, perhaps there is a better overall solution?

Reply:
Since this is working around branded assertions, could probably move this to webidl so that's the only shared dependency.

Reply:
The problem would exist before the branded assertions were added so it's not webidl related. When inspecting the prototype it would evaluate the properties, but now it doesn't.

Reply:
Sure custom inspect can throw by design but it's not applicable for any classes in say the Deno namespace like `Deno.File` where we don't configure the prototype to throw.

Reply:
It's not webidl specific or about configuring the prototype to throw though. Errors may occur since `this` does not refer to the instance of the class. Here's an isolated example that shows what the code was like before:

```ts
class MyClass {
  #myProperty = 5;

  get myProperty() {
    return this.#myProperty;
  }
  
  [Symbol.for("Deno.customInspect")](inspect) {
    const inner = {
      myProperty: this.myProperty
    };
    return `MyClass ${inspect(inner)}`;
  }
}

Deno.inspect(new MyClass()); // ok
Deno.inspect(MyClass.prototype); // throws
```

Also, I believe the console extension crate is necessary to inspect.

Reply:
>  Here's an isolated example that shows what the code was like before:

Right privates are a thing now, that tracks.

> Also, I believe the console extension crate is necessary to inspect.

Since we're removing the distinct inspect symbol, more of an abstract contract.

Anyway, here in the console crate is fine as-well just suggested it since you asked 😁